camel/memories/records.py

```
@@ -73,31 +75,56 @@ def from_dict(cls, record_dict: Dict[str, Any]) -> "MemoryRecord":
         message_cls = cls._MESSAGE_TYPES[record_dict["message"]["__class__"]]
         kwargs: Dict = record_dict["message"].copy()
         kwargs.pop("__class__")
+        # Restore enums expected by BaseMessage
+        if "role_type" in kwargs and isinstance(kwargs["role_type"], str):
+            try:
+                kwargs["role_type"] = RoleType(kwargs["role_type"])
+            except Exception as e:
+                logging.info(f"Failed to restore enum {kwargs['role_type']} for {message_cls.__name__}, {e}")
```

bare pass could mask real issues, maybe we can change to:
  - Use logging.warning instead of logging.info
  - Handle ValueError specifically instead of generic Exception
  - Consider keeping the string value or providing a default instead of silently passing